spec/controllers/external_users/registrations_controller_spec.rb

```diff
@@ -6,122 +6,115 @@
   end
 
   describe 'POST #create' do
-    let(:email) { Faker::Internet.email }
+    subject(:perform_post) { post :create, params: { user: sign_up_attributes } }
+
     let(:sign_up_attributes) do
-      {
-        first_name: 'Bob',
+      { first_name: 'Bob',
         last_name: 'Smith',
-        email: email,
+        email: 'foo@bar.com',
         password: 'password1234',
-        password_confirmation: 'password1234'
-      }
+        password_confirmation: 'password1234',
+        terms_and_conditions: '1' }
     end
 
-    context 'when env not api-sandbox' do
-      before { post :create, params: { user: sign_up_attributes, terms_and_conditions_acceptance: '1' } }
-
-      it 'redirects to user sign up path' do
-        expect(response).to redirect_to(external_users_root_url)
-      end
-
-      it 'does not create a user' do
-        expect(User.count).to eq(0)
-      end
-
-      it 'does not create an external user' do
-        expect(ExternalUser.count).to eq(0)
+    context 'when env not api-sandbox but attributes valid' do
+      around do |example|
+        with_env('production') do
+          example.run
+        end
       end
 
-      it 'does not create a provider' do
-        expect(Provider.count).to eq(0)
-      end
+      it { expect(perform_post).to redirect_to(external_users_root_url) }
+      it { expect { perform_post }.to change(User, :count).by(0) }
+      it { expect { perform_post }.to change(ExternalUser, :count).by(0) }
+      it { expect { perform_post }.to change(Provider, :count).by(0) }
```

Comment:
nit

I'd prefer

```ruby
expect { perform_post }.not_to change(User, :count)
etc
```

(but that is probably a matter of opinion)

Reply:
I generally prefer truthy statements, but that is probably a habit that comes from seeing false positives with negative statements like `.not_to have_content('blah')` when the page errored for unrelated reasons